hammerblade/torch/kernel/kernel_cat.cpp

```diff
@@ -0,0 +1,44 @@
+//====================================================================
+// _cat kernel
+//====================================================================
+// simple _cat kernel only works with 0 dim
+//
+// Authors : Lin Cheng, Janice Wei
+// Date    : 07/29/2020, 08/04/2020
+
+
+#include <kernel_common.hpp>
+
+extern "C" {
+
+//====================================================================
+// tensorlib__cat
+//====================================================================
+// This is a simple _cat kernel only works with 0 dim
+
+__attribute__ ((noinline))
+int tensorlib__cat( hb_tensor_t** tensors_p, hb_tensor_t* result_p,
+                    uint32_t* length_p, int32_t* dim_p)
+{
+  HBTensor<float> result(result_p);
+  uint32_t length = *length_p;
+  int32_t dim = *dim_p;
+  HBTensor<float> tensor(tensors_p[0]);
+  int32_t size = tensor.numel();
+  bsg_cuda_print_stat_kernel_start();
+
+  hb_tiled_for(result.numel(), [&] (size_t i) {
+    size_t j = i / size;
+    size_t index = i % size;
```

Comment:
Yeah here is an assumption that isn't always true.

But as a first step, it is ok to fail on the case where input tensors have different length. However, it cannot fail silently -- you would want to check for it and throw an error